stock_picking_batch_extended/tests/test_batch.py

```diff
@@ -0,0 +1,440 @@
+            "warehouse_id": warehouse,
+        }
+        procurements = [
+            self.env["procurement.group"].Procurement(
+                self.product11,
+                1,
+                self.env.ref("uom.product_uom_unit"),
+                self.customer_loc,
+                "test",
+                "TEST",
+                warehouse.company_id,
+                values,
+            )
+        ]
+        group.run(procurements)
+        pickings = self.picking_model.search([("group_id", "=", group.id)])
+        self.assertEqual(2, len(pickings))
+        picking = pickings.filtered(lambda p: p.state == "assigned")
+        picking.move_line_ids[0].qty_done = 1
+        picking.action_put_in_pack()
+        other_picking = pickings.filtered(lambda p: p.id != picking.id)
+        picking._action_done()
+        self.assertEqual("assigned", other_picking.state)
+        # We add the 'package' picking in batch
+        other_picking.batch_id = self.batch
+        self.batch.action_assign()
+        context = self.batch.action_transfer()
+        Form(
+            self.env["stock.immediate.transfer"].with_context(context)
+        ).save().process()
+        self.assertEqual("done", self.batch.state)
+        self.assertEqual("assigned", self.picking.state)
+        self.assertEqual("assigned", self.picking2.state)
+        self.assertEqual("assigned", other_picking.state)
+
+    def test_remove_undone(self):
+        self.picking2.action_cancel()
+        self.assertEqual("assigned", self.picking.state)
+        self.assertEqual("cancel", self.picking2.state)
+        self.assertEqual("draft", self.batch.state)
+
+        self.batch.remove_undone_pickings()
+
+        self.assertEqual("cancel", self.batch.state)
+        self.assertEqual(1, len(self.batch.picking_ids))
+        self.assertEqual(self.picking2, self.batch.picking_ids)
+
+    def test_partial_done(self):
+        # If user filled some quantity_done manually in operations tab,
+        # we want only these qties to be processed.
+        # So picking with no qties processed are release and backorder are
+        # created for picking partially processed.
+        self.batch.action_assign()
+        self.assertEqual("assigned", self.picking.state)
+        self.assertEqual("assigned", self.picking2.state)
+        self.picking.move_line_ids[0].qty_done = 1
+        context = self.batch.action_transfer().get("context")
+        # confirm transfer action creation
+        # Inmediate transfer? action
+        inmediate_transfer_action = (
+            Form(self.env["stock.immediate.transfer"].with_context(context))
+            .save()
+            .process()
+        )
+        backorder_confirmation_context = inmediate_transfer_action.get("context")
+        # Create backorder? action
+        Form(
+            self.env["stock.backorder.confirmation"].with_context(
+                backorder_confirmation_context
+            )
+        ).save().process()
```

Comment:
@sonhd91 to adapt/fix the unit test falling I suggest:
```suggestion
        self.picking2.move_line_ids[0].qty_done = 1
        self.picking2.move_line_ids[1].qty_done = 1
        context = self.batch.action_transfer().get("context")
        # Create backorder? action
        Form(
            self.env["stock.backorder.confirmation"].with_context(
                context
            )
        ).save().process()
```

Reply:
Thanks for review, I just update the branch